app/models/pet.rb

```diff
@@ -1,6 +1,8 @@
 class Pet < ApplicationRecord
 
   has_many :pet_images, dependent: :destroy
+  has_many :pet_areas, dependent: :destroy
+  has_many :areas, :through => :pet_areas
```

Comment:
記法はなるべく統一しておいた方が良いかなと思いました。
```suggestion
  has_many :areas, through: :pet_areas
```